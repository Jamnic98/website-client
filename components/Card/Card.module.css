.card {
  display: flex;
  flex-direction: row;
  background-color: white;
  box-shadow: var(--elevated-content);
  max-width: 600px;
}

.card:hover .img {
  transform: scale(1.15);
}

.card:hover .title {
  color: var(--c-text-dark);
}

.title {
  font-size: 1.2em;
  font-weight: bold;
  margin-bottom: 0.5em;
}

.imgContainer {
  min-width: 140px;
  min-height: 140px;
  overflow: hidden;
  position: relative;
}

.textContainer {
  box-sizing: border-box;
  position: relative;
  text-align: left;
  color: var(--c-link);
  padding: 1em;
  line-height: 1em;
}

.description {
  color: var(--c-text-dark);
}

@media screen and (max-width: 400px) {
  .imgContainer {
    min-width: 100px;
    min-height: 100px;
    overflow: hidden;
    position: relative;
  }

  .textContainer {
    font-size: 0.7em;
    line-height: 1.2em;
    padding: 0.8em;
  }
}
