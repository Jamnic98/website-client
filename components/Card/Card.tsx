import Link from 'next/link';
import Image from 'next/image';
import React, { useState } from 'react';
import styles from './Card.module.css';

interface Props {
  cardObj: {
    title: string;
    description: string;
    imageURI: string;
    URI: string;
  };
}

function Card(props: Props) {
  const [isFocused, setIsFocused] = useState(false);
  const setFocus = () => {
    setIsFocused(!isFocused);
  };

  const { title, description, imageURI, URI } = props.cardObj;
  return (
    <Link href={URI}>
      <a
        onMouseOver={() => setFocus()}
        onMouseOut={() => setFocus()}
        data-testid='card'
        className={`${styles.card} ${isFocused ? `${styles.focused}` : ``}`}
      >
        <span className={styles.imgContainer}>
          <Image
            src={imageURI}
            alt=''
            className={`${styles.img} ${isFocused ? `${styles.focused}` : ``}`}
            layout='fill'
          />
        </span>
        <span
          className={`${styles.textContainer} ${
            isFocused ? `${styles.focused}` : ``
          }`}
          data-testid='textContainer'
        >
          <div className={styles.title}>{title}</div>
          <div className={styles.description}>{description}</div>
        </span>
      </a>
    </Link>
  );
}

export default Card;
