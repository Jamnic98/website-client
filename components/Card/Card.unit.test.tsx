import Card from './Card';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('Card', () => {
  it('has correct classes', () => {
    const testObj = {
      title: 'Card',
      description: 'Test card',
      imageURI: '/',
      linkURL: '/',
    };
    render(<Card cardObj={testObj} />);
    const card = screen.getByTestId('card');
    expect(card).toHaveClass('card');

    fireEvent.mouseOver(card);
    expect(card).toHaveClass('card focused');

    fireEvent.mouseOut(card);
    expect(card).toHaveClass('card');
  });
});
