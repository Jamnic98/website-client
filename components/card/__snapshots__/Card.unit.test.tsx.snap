// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders Card unchanged 1`] = `
<a
  className="card"
  data-testid="application-card"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  onTouchStart={[Function]}
>
  <span
    className="imgContainer"
  >
    <img
      alt="Card Title"
      className="img"
      data-nimg="1"
      decoding="async"
      height={140}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2F&w=384&q=75"
      srcSet="/_next/image?url=%2F&w=256&q=75 1x, /_next/image?url=%2F&w=384&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={140}
    />
  </span>
  <div
    className="textContainer"
  >
    <div
      className="title"
    >
      Card Title
    </div>
    <div
      className="description"
    >
      Test description.
    </div>
  </div>
</a>
`;
